### PR TITLE
Feature/dapp 543

### PR DESCRIPTION
**Pull request by NickNish09**
# Description

- Add new payments page
- Refactor payments page to be generic
- Add it to non profits flow (in the next PRs the idea is to migrate all pages to use it)
## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Dangerous changes
- [ ] Chore (change that does not effect how the code works, eg: change color)
- [ ] Tests

## How to test

Put here all the things need to test this PR and verify your changes, also list any relevant details for tests (eg: have a wallet)

- Test A

### CODE REVIEW

File: src/pages/promoters/PaymentPage/CardInfoSection/styles.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  div:first-of-type {
+    margin-right: 8px;
+  }
+`;
+
+export const Form = styled.form`
+  input {
+    height: 48px;
+    border: 1px solid ${({ theme }) => theme.colors.orange40};
+    font-weight: 600;
```
Comment (yangalli) [inline src/pages/promoters/PaymentPage/CardInfoSection/styles.ts:29]: I think we don't need to use font-weight here :)

File: src/pages/promoters/PaymentPage/index.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+import { Currencies } from "types/enums/Currencies";
+import { useEffect, useState } from "react";
+import { useCardPaymentInformation } from "contexts/cardPaymentInformationContext";
+import NonProfit from "types/entities/NonProfit";
+import * as S from "./styles";
+import UserInfoSection from "./UserInfoSection";
+import CardInfoSection from "./CardInfoSection";
+import SupportImage from "../SupportCausePage/assets/support-image.png";
+
+type LocationState = {
```
Comment (yangalli) [inline src/pages/promoters/PaymentPage/index.tsx:17]: If you want, you can use the context here to save cause and nonProfit.

File: src/pages/promoters/PaymentPage/styles.ts [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+`;
+
+export const Title = styled.h1`
+  ${defaultParagraphLarge};
+  text-align: center;
+`;
+
+export const TitleHighlight = styled.span`
+  ${defaultSubtitleLarge};
+  font-weight: 600;
```
Comment (yangalli) [inline src/pages/promoters/PaymentPage/styles.ts:44]: ditto

**Review by yangalli — APPROVED** [review]

File: src/pages/promoters/PaymentPage/index.tsx (hunk 2 above)
Comment (NickNish09) [reply src/pages/promoters/PaymentPage/index.tsx:17]: As we talked, just to register in the PR:
I think that it is good to let this page receiving this param, and if we want to make a payment from any page we just pass this params.
The previous page don`t have to know about the context

**Pull request merged**